data-prepper-plugins/common/src/main/java/org/opensearch/dataprepper/plugins/source/RandomStringSource.java

```diff
@@ -6,69 +6,68 @@
 package org.opensearch.dataprepper.plugins.source;
 
 import org.opensearch.dataprepper.model.annotations.DataPrepperPlugin;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPluginConstructor;
 import org.opensearch.dataprepper.model.buffer.Buffer;
 import org.opensearch.dataprepper.model.event.Event;
 import org.opensearch.dataprepper.model.event.JacksonEvent;
 import org.opensearch.dataprepper.model.record.Record;
 import org.opensearch.dataprepper.model.source.Source;
-import com.google.common.util.concurrent.ThreadFactoryBuilder;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.util.UUID;
-import java.util.concurrent.ExecutorService;
-import java.util.concurrent.Executors;
-import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
 
 /**
  * Generates a random string every 500 milliseconds. Intended to be used for testing setups
  */
-@DataPrepperPlugin(name = "random", pluginType = Source.class)
+@DataPrepperPlugin(name = "random", pluginType = Source.class, pluginConfigurationType = RandomStringSourceConfig.class)
 public class RandomStringSource implements Source<Record<Event>> {
-
     private static final Logger LOG = LoggerFactory.getLogger(RandomStringSource.class);
+    private static final int BUFFER_WAIT = 500;
+    private final long waitTimeInMillis;
 
-    private ExecutorService executorService;
     private volatile boolean stop = false;
+    private Thread thread;
 
-    private void setExecutorService() {
-        if(executorService == null || executorService.isShutdown()) {
-            executorService = Executors.newSingleThreadExecutor(
-                    new ThreadFactoryBuilder().setDaemon(false).setNameFormat("random-source-pool-%d").build()
-            );
-        }
+    @DataPrepperPluginConstructor
+    public RandomStringSource(final RandomStringSourceConfig config) {
+        waitTimeInMillis = config.getWaitDelay().toMillis();
     }
 
     @Override
     public void start(final Buffer<Record<Event>> buffer) {
-        setExecutorService();
-        executorService.execute(() -> {
+        if(thread != null) {
+            throw new IllegalStateException("This source has already started.");
+        }
+
+        thread = new Thread(() -> {
```

Comment:
QUES: what is the benefit of this change?

Reply:
The current code doesn't need a threat pool because it runs one thread. Taking this approach simplifies some of the branching in the code, making it easier to get higher test coverage.